AI summarizer

jacopobellosi opened the pull request:
let me know what do you think

AriYusa committed:
initial set up

Deyvla2 committed:
Initial setup for user management microservice

Deyvla2 committed:
Add Docker support for user management microservice

Deyvla2 committed:
Add requirements.txt for pip users

AriYusa committed:
rename to character-recognition

AriYusa committed:
update pre-commit

AriYusa committed:
set up flask

AriYusa committed:
edit github action

AriYusa committed:
test ci

AriYusa committed:
set up docker

AriYusa committed:
add simple ocr

patpat261 committed:
initial setup

AriYusa committed:
Merge branch 'main' into character-recognition/mvp

# Conflicts:
#	.gitignore
#	README.md

AriYusa committed:
Delete unnecessary files

AriYusa committed:
Merge pull request #2 from jacopobellosi/character-recognition/mvp

Character recognition/mvp

JulieL999 committed:
Created the first version of the homescreen

AmirElkess committed:
initial commit

Initial commit with basic working audio transcription

Backend: FastAPI
Model: Vosk (lightweight English model)

AmirElkess committed:
Merge pull request #3 from jacopobellosi/transcriptify

initial commit

patpat261 committed:
implement paraphrasing feature

AriYusa commented:
It is not necessary to create PR when merging from the master into your branch. Only the other way, when you merge your branch into master

AmirElkess commented:
You can add a rule for `.DS_Store` in `.gitignore` then remove it from main branch, as it is a macOS-only filesystem file that is not relevant here.

Also, a good place to start is the [GitHub Python .gitignore template](https://github.com/github/gitignore/blob/main/Python.gitignore) which we can directly use and it contains most exclusion rules that we might need in a Python project.

AmirElkess committed:
create a readme file

Added README file detailing key libraries used, instructions for running the application, and a TODO list for future tasks.

AmirElkess committed:
update .gitignore

Updated .gitignore using the GitHub Python template.

AmirElkess committed:
File Conversion, UUID-based Naming, and Auto-deletion for Vosk Compatibility and Storage Optimization

Implemented logic for converting files to a standardized format compatible with the Vosk library using FFmpeg.

Integrated UUIDv1-based file naming to ensure collision-free file handling.

Implemented automated deletion of uploaded/converted audio resources to optimize storage.

AmirElkess reviewed (approved):
Approved.

patpat261 committed:
Merge branch 'main' of https://github.com/jacopobellosi/ASE---AI-summarizer into paraphrasing-tool

JulieL999 committed:
Created navigation

JulieL999 committed:
Create the first version of Paraphrize page

Deyvla2 committed:
File-Restructuring

Deyvla2 committed:
File Restructring

JulieL999 committed:
Improved the paraphrizer

JulieL999 committed:
Created the summarizer

JulieL999 committed:
Add home button

AmirElkess committed:
Merge pull request #5 from jacopobellosi/transcriptify

Transcriptify

AmirElkess committed:
Adding Poetry and Black

Added Poetry for package management, and Black for code formatting.

AmirElkess committed:
code formatting and readme fix

AmirElkess committed:
Merge pull request #6 from jacopobellosi/transcriptify

Transcriptify

AriYusa reviewed (changes requested):
Jacopo, can you rename the service? Because now it doesn't reflect functionality - summarization and it is recommended to use `-` instead of spaces and use lowercase

patpat261 committed:
added dockerfile

AriYusa committed:
add postprocessing

AriYusa committed:
create docker-compose for working services

patpat261 committed:
remove .env from version control

patpat261 committed:
fix dockerfile

patpat261 committed:
Merge pull request #7 from jacopobellosi/paraphrasing-tool

Paraphrasing tool

AmirElkess committed:
dockerizing the microservice

Dockerized the application.
Removed the FFmpeg executable from the application files; it now needs to be installed on the system (not needed for the Docker build).

AmirElkess committed:
Merge pull request #9 from jacopobellosi/transcriptify

dockerizing the microservice

AmirElkess committed:
Update readme.md

AmirElkess committed:
Merge pull request #10 from jacopobellosi/transcriptify

Update readme.md

AriYusa committed:
add docker-compose

AriYusa committed:
dockerfile for 3 services

AriYusa committed:
update gitignore

JulieL999 committed:
Tried to connect to Summarizer server.

JulieL999 committed:
Merge pull request #12 from jacopobellosi/frontend

Frontend

AriYusa committed:
Merge pull request #13 from jacopobellosi/integtrate

Integtrate

Deyvla2 committed:
Pre-Commit Flake8 Removal

Deyvla2 committed:
Project Files Update

Deyvla2 committed:
Database API implementation

Deyvla2 committed:
Database Implementation

AmirElkess committed:
Adding endpoint testing results to readme.md

AmirElkess committed:
Merge pull request #15 from jacopobellosi/transcriptify

Adding endpoint testing results to readme.md

AmirElkess committed:
fix nested frontend directory structure

AmirElkess committed:
Add initial content for About Us and Documentation pages.

AmirElkess committed:
Minor tweaks to the UI

AmirElkess committed:
Merge pull request #16 from jacopobellosi/frontend

Restructure Frontend Directory and Add Initial Content for About Us and Documentation Pages

AmirElkess committed:
Add CORS Middleware and Connect Frontend Upload Audio to Backend

Added Cross-Origin Resource Sharing (CORS) middleware to the voice-transcription microservice to resolve browser security issues.
Integrated the "Upload Audio" button in the frontend with the backend API endpoint.
Included Arina's code from the frontend-upload-image branch as-is. Arina, please test and confirm that the functionality works as expected.

AriYusa committed:
Refactor to parent processor

AriYusa committed:
add dockerfile

AriYusa committed:
support CORP

AriYusa committed:
change reqest links to container names

AriYusa committed:
new text as new paragraph

AmirElkess committed:
Merge pull request #17 from jacopobellosi/frontend-upload-files

Add CORS Middleware and Connect Frontend Upload Audio to Backend

AriYusa committed:
test from

jacopobellosi committed:
authentication page

JulieL999 committed:
New version of the frontend main page

JulieL999 committed:
Integrated image input

AriYusa committed:
add openAPI for paraphrasing-tool

AriYusa committed:
add quiz page

AriYusa committed:
add poetry

AriYusa committed:
generate api

AriYusa committed:
parce test

AriYusa committed:
move openapi.yaml inside microservice and generate service

AriYusa committed:
edit dockerfile to support poetry

AriYusa committed:
failed attempt to run it

AriYusa committed:
yet another not working atempt

AriYusa committed:
remove open api

AriYusa committed:
add request to fronend

AriYusa committed:
handle error in generation

AriYusa committed:
edit frontend

AriYusa committed:
add test

AriYusa committed:
clean up

AriYusa committed:
UI improvements

AriYusa committed:
improve quiz generation

JulieL999 committed:
Add functionality to the buttons Paraphrase, Summarise, Upload Audio and Image

AmirElkess committed:
Update readme.md

with steps on running the application

AmirElkess committed:
Initial API-Gateway setup

Set up Nginx as the API gateway

- Added nginx.conf with a sample API endpoint for the voice-transcription microservice.
- Created endpoints.md to document all available API endpoints.

AmirElkess committed:
Update nginx.conf to include all endpoints for the transcription microservice

jacopobellosi committed:
update

AriYusa committed:
Merge pull request #19 from jacopobellosi/exam_prep_no_openapi

Quiz generator

AmirElkess committed:
update nginx.conf to include config for OCR microservice

AmirElkess committed:
minor edits

AmirElkess committed:
update nginx.conf for paraphrase microservice

jacopobellosi committed:
color update

jacopobellosi committed:
Merge branch 'main' into frontend

jacopobellosi committed:
Merge pull request #20 from jacopobellosi/frontend

login/registration Frontend

jacopobellosi committed:
Merge pull request #21 from jacopobellosi/AI-summarizer-update

update AI summarier

JulieL999 committed:
Add new note

Deyvla2 committed:
Changes for merge to main

AriYusa committed:
fix dockerfile

AriYusa committed:
Merge branch 'main' of https://github.com/jacopobellosi/ASE---AI-summarizer into user-management

Deyvla2 committed:
Delete .gitignore

Deleted .gitignore for merge to main

AriYusa committed:
Merge branch 'user-management' of https://github.com/jacopobellosi/ASE---AI-summarizer into user-management

Deyvla2 committed:
Merge branch 'user-management' of https://github.com/jacopobellosi/ASE---AI-summarizer into user-management

AmirElkess committed:
Dockerizing nginx api-gateway

Dockerizing API-Gateway

- readme.md with instructions on how to run the docker image
- adding nginx container definition to the main project docker-compose file

endpoints tested for the voice transcription microservice, accessible on
POST `http://localhost:5555/api/v1/transcription/transcribe/` <br>
GET `http://localhost:5555/api/v1/transcription/`

AriYusa committed:
ignore vscode setting

AriYusa committed:
Merge branch 'user-management' of https://github.com/jacopobellosi/ASE---AI-summarizer into user-management

Deyvla2 committed:
Merge pull request #14 from jacopobellosi/user-management

User management Micro-service Implementation

AriYusa committed:
fix

AriYusa committed:
Merge pull request #22 from jacopobellosi/fix_ignore

fix

AmirElkess committed:
modify nginx for character recognition microservice

modified nginx.conf and docker-compose.yaml

AmirElkess committed:
modify nginx for paraphrasing microservice

jacopobellosi committed:
API for registration

AmirElkess committed:
Merge pull request #23 from jacopobellosi/api-gateway

API Gateway

AmirElkess committed:
unit/endpoint testing initial setup

AmirElkess committed:
testing all endpoints

Tested:
- `GET /`
- `POST /transcribe` with an audio file

Added 2 audio test files to be committed in git for CI, `.gitignore` modified accordingly

AmirElkess committed:
minor modification to testing

jacopobellosi committed:
Login and registration API complete

jacopobellosi committed:
authentication with profile page

Now when a user login/register there is a token for the authentication.
If authenticated can access a personal profile page

AmirElkess committed:
making changes to support github CI

AmirElkess committed:
Merge pull request #24 from jacopobellosi/Testing-VoiceTranscription

Testing voice transcription endpoints

AmirElkess committed:
added an extra transcribe/ endpoint test with a different test file

AmirElkess committed:
Create voice-transcription-ci.yml

AmirElkess committed:
Merge pull request #25 from jacopobellosi/voice-transcription-CI

Create voice-transcription-ci.yml

AmirElkess committed:
Update readme.md

AmirElkess committed:
Merge pull request #26 from jacopobellosi/Testing-VoiceTranscription

Testing CI pipeline for voice-transcription microservice

patpat261 committed:
fix some error from frontend

patpat261 committed:
fix import statement refering to a parent package

patpat261 committed:
added new feature(notes)

patpat261 committed:
Pulled frontend changes from new_version_of_frontend

patpat261 committed:
allow port 3000 for access in frontend

patpat261 committed:
integrated the functions to frontend

AriYusa committed:
pull main

AmirElkess committed:
replacing boring-text-editor with rich-text-editor

AriYusa committed:
add test generate into magic options

patpat261 committed:
updated the dependencies in pyproject.toml file, and also fix dockerfile

AmirElkess committed:
support adding text programmatically to rich-text-editor (transcribe-text)

AmirElkess committed:
minor UI tweak

AmirElkess committed:
make text result non-editable

AmirElkess committed:
for summarise function: convert html text to markdown (so backend can deal with it)

AriYusa committed:
fix sizes

AriYusa committed:
pretify

patpat261 committed:
revert imported library path

patpat261 committed:
docker works

patpat261 committed:
docker composee includes user management

patpat261 committed:
Merge pull request #31 from jacopobellosi/user-management-notes

User management notes with docker

jacopobellosi committed:
update homescreen with API

AriYusa committed:
resolve conflict

AmirElkess committed:
Merge pull request #30 from jacopobellosi/quiz_into_frontend

New frontend

patpat261 committed:
added update and delete for notes

patpat261 committed:
Merge branch 'user-management-notes' of https://github.com/jacopobellosi/ASE---AI-summarizer into user-management-notes

patpat261 committed:
Merge branch 'main' into user-management-notes

patpat261 committed:
Merge pull request #32 from jacopobellosi/user-management-notes

User management notes

AriYusa committed:
Add saving

AriYusa committed:
Merge pull request #33 from jacopobellosi/integrate_note_update

Add saving

jacopobellosi committed:
Fixed authentication pages

jacopobellosi committed:
Merge pull request #34 from jacopobellosi/user-management-notes

Fixed authentication pages

AmirElkess committed:
minor fixes

- Add descriptive container_name for all docker containers in docker-compose
- UI fix: give 'clear input button' a transparent background

AmirElkess committed:
refactor summarizer directory structure

AmirElkess committed:
add CORS to summarizer

- add CORS to summarizer service
- modified pull-llama3.sh (to file line endings)

AmirElkess committed:
merging docker-compose containers from summarizer into "The main docker-compose"

AmirElkess committed:
Extract response from returned object from summarizer microservice

AmirElkess committed:
Merge pull request #35 from jacopobellosi/fixes-amir

Integrate summarizer with frontend, and merge its docker-compose with main docker-compose

AriYusa committed:
resolve conflict

AriYusa committed:
Merge pull request #8 from jacopobellosi/character-recognition/feature_improvements

add postprocessing

AmirElkess committed:
Update readme.md

AmirElkess committed:
Merge pull request #36 from jacopobellosi/AmirElkess-patch-1

Update readme.md

kappo250 committed:
docs: rewrite README with architecture overview and tech stack

kappo250 committed:
chore: add MIT license

jacopobellosi committed:
Merge pull request #37 from jacopobellosi/polish/readme-and-license

docs: add proper README and MIT license
